Allow resizing two column layout

maier49 opened the pull request:
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit tests are included in the PR
* [ ] For new widgets, an entry has been added to the `.dojorc`
* [ ] For new widgets, `theme.variant()` is added to the root domnode
* [ ] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [ ] WidgetProperties are exported

<!--
Our bots should ensure:

* [ ] All contributors have signed a CLA
* [ ] The PR passes CI testing
* [ ] Code coverage is maintained
* [ ] The PR has been reviewed and approved
-->

**Description:**
Adds the ability to resize the two column layout
Resolves #1479

tomdye commented on src/theme/default/two-column-layout.m.css:
```diff
@@ -4,3 +4,7 @@
 }
```
Can you add some comments to these classes please and initial styling so that it works without a theme

tomdye commented:
@maier49 I've noticed that the mouse and the resize bar get out of sync when the scroll hits it's min / max. Do you think you could take a look at that.
With regards to the thumb for resizing, I'm not sure it _is_ the best way forward. Looks like others, including this simple react example use a border-left/right that animate in on hover instead? This would likely work better with other themes too and is less obtrusive.

 http://react-split-pane-v2.surge.sh/?SimpleExample

maier49 commented:
@tomdye I think I've fixed that issue. It was because I was adding to delta to the actual element size instead of the cached size. So past the limit it wouldn't track those changes really and then when moving back it would shrink the element.

I've also updated the styles.

tomdye commented:
@maier49 do you think you could add a longer delay to the wider drag bar transition and also ensure that it stays visible when the bar is focused/active as wall as when hovered. Could you also increase the z-index of the divider line so that it shows above the wider hover border.

maier49 commented:
@tomdye We don't have a delay on transitions anywhere else so I added a new variable for that, set it to .5s. I changed the hover style to also include focus/active states. And there isn't really a relevant z-index because it's just the background-color vs border-colors of the but I did adjust it so that the line is still visible.
